### PR TITLE
Telemetry for Actors & Services framework

### PR DESCRIPTION
**Pull request by vshan**
- This PR adds Telemetry for the Actors & Services framework through DCA via EventSource events.

### CODE REVIEW

File: src/Microsoft.ServiceFabric.Services/TelemetryConstants.cs [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+// Licensed under the MIT License (MIT). See License.txt in the repo root for license information.
+// ------------------------------------------------------------
+
+namespace Microsoft.ServiceFabric.Services
+{
+    internal static class TelemetryConstants
+    {
+        // Names of all the telemetry events.
+        internal static readonly string ServiceLifecycleEventName = "TelemetryEvents.ServiceLifecycleEvent";
+        internal static readonly string ActorStateProviderUsageEventName = "TelemetryEvents.ActorStateProviderUsageEvent";
```
Comment (amanbha) [inline src/Microsoft.ServiceFabric.Services/TelemetryConstants.cs]: >TelemetryEvents [](start = 76, length = 15)

Please move Actor related constants to Actors project.

File: src/Microsoft.ServiceFabric.Services/ServiceTelemetry.cs [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+// Licensed under the MIT License (MIT). See License.txt in the repo root for license information.
+// ------------------------------------------------------------
+
+namespace Microsoft.ServiceFabric.Services
+{
+    using System.Fabric;
+
+    /// <summary>
+    /// ServiceTelemetry contains the telemetry methods for ServiceFramework.
+    /// </summary>
```
Comment (amanbha) [inline src/Microsoft.ServiceFabric.Services/ServiceTelemetry.cs]: Add in the summary that this class is to be used by other Service Fabric  components only.

File: src/Microsoft.ServiceFabric.Services/Communication/Client/ServicePartitionClient.cs [hunk 3]
```diff
@@ -59,6 +59,11 @@ public class ServicePartitionClient<TCommunicationClient> : IServicePartitionCli
             this.targetReplicaSelector = targetReplicaSelector;
             this.lastRsp = null;
             this.retrySettings = retrySettings ?? new OperationRetrySettings();
+
+            ServiceTelemetry.CustomCommunicationClientUsageEvent(
```
Comment (amanbha) [inline src/Microsoft.ServiceFabric.Services/Communication/Client/ServicePartitionClient.cs]: @suchiagicha  Could you please help whats the  best way to identify a custom communication client as this is called by Remoting client as well.

File: src/Microsoft.ServiceFabric.Services/ServiceTelemetry.cs (hunk 2 above)
Comment (vshan) [reply src/Microsoft.ServiceFabric.Services/ServiceTelemetry.cs]: Thanks, I have added that to the summary.

File: src/Microsoft.ServiceFabric.Services/TelemetryConstants.cs (hunk 1 above)
Comment (vshan) [reply src/Microsoft.ServiceFabric.Services/TelemetryConstants.cs]: Thanks, have moved those constants.

File: src/Microsoft.ServiceFabric.Services/Communication/Client/ServicePartitionClient.cs (hunk 3 above)
Comment (vshan) [reply src/Microsoft.ServiceFabric.Services/Communication/Client/ServicePartitionClient.cs]: Perhaps I should rename this to "CheckCustomCommunicationClientUsageEvent" as internally there is a check, and telemetry is only logged for non-remoting case.

Comment (suchiagicha) [reply src/Microsoft.ServiceFabric.Services/Communication/Client/ServicePartitionClient.cs]: @vshan  User creates ServicePartitionClient for each call , which might be too much of data. Discussed with @amanbha  , we dont't need any client side telemetry. Service/Listener side telemetry should be enough.

Comment (vshan) [reply src/Microsoft.ServiceFabric.Services/Communication/Client/ServicePartitionClient.cs]: @amanbha  @suchiagicha  Ok sure, I have removed client side telemetry for ServicePartitionClient. Thanks

**Comment by juho-hanhimaki** [comment]
I feel this issue might be relevant to work that is being done in this PR: https://github.com/microsoft/service-fabric-services-and-actors-dotnet/issues/179

**Comment by amanbha** [comment]
@juho-hanhimaki  This is different from #179.

**Pull request merged**